components/places/ios/Places/LoginRecord.swift

```diff
@@ -0,0 +1,182 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import Foundation
+/*
+open class LoginRecord {
+    /// The guid of this record. When inserting records, you should set this
+    /// to the empty string. If you provide a non-empty one to `add`, and it
+    /// collides with an existing record, a `LoginsStoreError.DuplicateGuid`
+    /// will be emitted.
+    public var id: String
+
+    /// This record's hostname. Required. Attempting to insert
+    /// or update a record to have a blank hostname, will result in a
+    /// `LoginsStoreError.InvalidLogin`.
+    public var hostname: String
+
+    /// This record's password. Required. Attempting to insert
+    /// or update a record to have a blank password, will result in a
+    /// `LoginsStoreError.InvalidLogin`.
+    public var password: String
+
+    /// This record's username, if any.
+    public var username: String? = nil
+
+    /// The challenge string for HTTP Basic authentication.
+    ///
+    /// Exactly one of `httpRealm` or `formSubmitURL` is allowed to be present,
+    /// and attempting to insert or update a record to have both or neither will
+    /// result in an `LoginsStoreError.InvalidLogin`.
+    public var httpRealm: String? = nil
```

Comment:
Maybe 
```
enum ChallengeString {
 case httpRealm(value: String)
 case formSubmitUrl(value: String)
}
```

Reply:
Sorry, this was copypasta, this is from logins!